external/fv3fit/setup.py

```diff
@@ -20,10 +20,17 @@
     "plotly",
     "tensorboard",
     "pandas",
+    "torch==1.12.0",
+    "torchvision==0.13.0",
+    "dgl==0.9.0",
+    "geographiclib==1.52",
+    "geopy==2.2.0",
+    "tensorflow-datasets",
```

Comment:
This is a lot of dependencies to add. Are they all needed? Could we avoid adding some if I can write the graph generation code manually? In any case, make sure to add all new dependencies to the PR description.

Reply:
Additionally, we'll have to run the `make lock_pip` command to update the constraints file and then document the licenses of the new packages.

Reply:
Is `tensorflow-datasets` needed? I thought we already had tensorflow installed.

Reply:
To merge this PR, we need to remove the pins on these package versions, run `make lock_pip`, and commit the updated `constraints.txt` file to the repo.

Do we depend on geographiclib explicitly, or only through geopy? If it's only an indirect dependence, you can remove it from this setup.py file

I would really like to remove this geopy dependency by writing graph generation which doesn't need it, but we can do it separately.